### PR TITLE
Pt35 register backend

### PR DESCRIPTION
**Pull request by Cesar-M-Diaz**
## Register backend

created the following functionalities :

- the register post request with a createUser controller
- the createUser controller filters the roles of the user (tutor or student)
- defined two schemas, one for tutor and one for user

### CODE REVIEW

File: models/student.model.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+const studentSchema = mongoose.Schema(
+  {
+    name: { type: String, required: [true, 'Name is required'] },
+    email: {
+      type: String,
+      match: /.+\@.+\..+/,
+      required: [true, 'Email is required'],
+      validate: {
+        validator: async function (value) {
```
Comment (leoneloliveros) [inline models/student.model.js:12]: Please view comments on login PR #3

File: controllers/register.controller.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  try {
+    const { type, inputs } = req.body;
+
+    let userSchema = '';
+
+    if (type === 'student') {
+      userSchema = Student;
+    } else {
+      userSchema = Tutor;
+    }
```
Comment (leoneloliveros) [inline controllers/register.controller.js:15]: Excellent implementation, you can refactor a little more but It's not necessary. 

`userSchema = (type === 'student') ? Student : Tutor`

**Review by leoneloliveros — COMMENTED** [review]
Great job refactoring code. Very simple and easy to understand

File: models/student.model.js (hunk 1 above)
Comment (Cesar-M-Diaz) [reply models/student.model.js:12]: Hi Leonel, just added some logic to fix this.

File: controllers/register.controller.js (hunk 2 above)
Comment (Cesar-M-Diaz) [reply controllers/register.controller.js]: Applied your advice, it looks more elegant with the ternary operator.

File: models/tutor.model.js [hunk 3]
```diff
@@ -11,8 +11,8 @@ const tutorSchema = mongoose.Schema(
       required: [true, 'Email is required'],
       validate: {
         validator: async function (value) {
-          const tutor = await Tutor.findOne({ email: value });
           const student = await Student.findOne({ email: value });
+          const tutor = await Tutor.findOne({ email: value });
```
Comment (leoneloliveros) [inline models/tutor.model.js:15]: Be aware with ciclyc dependency

Comment (Cesar-M-Diaz) [reply models/tutor.model.js:13]: Hi Leonel, just fixed this problem.

**Review by leoneloliveros — APPROVED** [review]
Excellent job preventing cyclic

**Pull request merged**